chapters/2.md

```diff
@@ -1,211 +1,170 @@
-
-Polymer/Web components aren't just one among many choices, they are the *only* choice. Because if most of the people who'll be trying to consume online content *can't do it through the browser*, then something else will fill that need. And that means, it doesn't matter if you know React. It doesn't matter if you know Angular. People won't be using a browser anyway.
-
-But more than that, I think Polymer/Web component based development can be an order of magnitude more effective than their JavaScript-framework based counterparts, and the distance between here and there is only three things.
+What is a template? Where have you seen the term in other areas of life? There are
+drawing templates, cookie templates and many more. How are these templates used?
+When I take a drawing template and I trace the edges provided by the template I
+always get the same resulting drawing afterward. For example, a template outlining
+a house will always result in a house. You won't draw a house four times followed
+by a giraffe (unless something odd is going on :-P). Templates in the polymer realm
+are conceptually the same. The `<template>` element in the polymer realm specifies
+an outline for the module content so that every time the custom element is seen
+in an HTML file the same content will be stamped in place of its name.
+
+TODO The template actually refers to template in transpiler terms. Include here?
+TODO Probably not good to load newbies with that. Should defer to later.
+
+## JavaScript at the Bottom of the Page
+```javascript
+  Polymer({
+    is: 'simple-element',
+    ...
+  });
+```
+Whaaaaaaat? Why is there JavaScript at the bottom of this HTML file? Don't worry.
+That seems a bit odd until you get used to it. The reason this JavaScript is in
+our HTML file is actually simple. In Polymer, what we're specifying in our html
+file is a custom element. However, custom elements that only allow us to specify
+the markup to use in place of that element are extremely limited. What we need is
+a manner to couple that markup with functionality. It's starting to make sense,
+right? The JavaScript is present in the HTML file because what we're defining
+is a custom element not just html. And that custom element has to be capable of
+having custom behavior otherwise there wouldn't be much of a point in having
+custom elements in the first place. We define our custom behavior in that JavaScript.
+
+### The Polymer Function
+You may have noticed that our element definition appears to be the input to a function
+called Polymer. First of all, good eye! The Polymer(...) function allows us to define
+the name of our custom element (this must match the id attribute present in the
+`<dom-module>` element), [element properties](https://www.polymer-project.org/1.0/docs/devguide/properties),
+and more.
+
+### But, Defining Styling, Markup and Logic in the Same File is a Code Smell!
+TODO Should this be included here or no?
```

Comment:
I think you can defer this. Right now, the goal would be to build a mental model of a component. Then, once that's comfortable, the question arises: is there a better way to compose these constituent pieces? So that could be it's own "let's look at a google element. why do they pull out behavior into a different file?" etc kinda chapter

Reply:
Awesome!
- [ ] Remove possible section